java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java

```diff
@@ -99,18 +106,23 @@ public MethodTreeImpl(
     this.returnType = returnType;
     this.simpleName = Preconditions.checkNotNull(simpleName);
     this.parameters = Preconditions.checkNotNull(parameters);
+    this.openParenToken = parameters.openParenToken();
+    this.closeParenToken = parameters.closeParenToken();
     this.block = block;
+    this.semicolonToken = semicolonToken;
     this.throwsClauses = Preconditions.checkNotNull(throwsClauses);
     this.defaultToken = null;
     this.defaultValue = null;
   }
 
-  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName) {
+  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName, SyntaxToken semicolonToken) {
     Preconditions.checkState(this.simpleName == null);
     this.returnType = returnType;
     this.simpleName = simpleName;
+    this.semicolonToken = semicolonToken;
 
     prependChildren((AstNode) returnType, (AstNode) simpleName);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png)org/sonar/java/model/declaration/MethodTreeImpl must not use com/sonar/sslr/api/AstNode[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)